src/internal.c

```diff
@@ -9356,11 +9389,125 @@ int CheckAvailableSize(WOLFSSL *ssl, int size)
     return 0;
 }
 
+#ifdef WOLFSSL_DTLS13
+static int GetDtls13RecordHeader(WOLFSSL* ssl, const byte* input,
+                                 word32* inOutIdx,
+                                 RecordLayerHeader* rh, word16 *size)
+{
+    byte epoch_bits, seq_lo, seq_hi;
+    word16 header_length;
+    byte new_seq_lo[4];
+    int seq_present;
+    int read_size;
+    int ret;
+
+    read_size = ssl->buffers.inputBuffer.length - *inOutIdx;
+
+    ret = dtls13_parse_unifed_record_layer(ssl,
+                                           input + *inOutIdx,
+                                           read_size,
+                                           size,
+                                           &header_length,
+                                           &seq_lo,
+                                           &seq_hi,
+                                           &seq_present,
+                                           &epoch_bits);
+    if (ret != 0)
+        return ret;
+
+    /* type is implicit in unified header  */
+    rh->type = application_data;
+    c16toa(*size, rh->length);
+
+    /* in unified header version is implicit */
+    rh->pvMajor = ssl->version.major;
+    rh->pvMinor = DTLSv1_2_MINOR;
+
+    if ((ssl->keys.dtls_epoch & EE_MASK) == epoch_bits) {
+        ssl->keys.curEpoch = ssl->keys.dtls_epoch;
+    } else if (((ssl->keys.dtls_epoch - 1) & EE_MASK) == epoch_bits) {
+        ssl->keys.curEpoch = ssl->keys.dtls_epoch - 1;
+    } else {
+        return SEQUENCE_ERROR;
+    }
+
+    ssl->keys.curSeq_hi = ssl->keys.peer_sequence_number_hi;
+
+    c32toa(ssl->keys.peer_sequence_number_lo, new_seq_lo);
+    new_seq_lo[3] = seq_lo;
+    if (seq_present)
+        new_seq_lo[2] = seq_hi;
+    ato32(new_seq_lo, &ssl->keys.curSeq_lo);
```

Comment:
Need to check if current epoch. Otherwise use `dtls_prev_sequence_number_hi` and `dtls_prev_sequence_number_lo`.

Reply:
seq number will take another approach in future commits